IND-5776 Postgres passwordless release test

raviharshicorp opened the pull request:
## Background

Added variables related to AWS postgres passwordless changes 

Relates : https://github.com/hashicorp/terraform-enterprise/pull/3079
https://github.com/hashicorp/terraform-aws-terraform-enterprise/pull/379

## How has this been tested?
CI/CD: https://github.com/hashicorp/terraform-enterprise/actions/runs/18079265431/job/51440243884

raviharshicorp committed:
Add AWS IAM database authentication support

- Add database_passwordless_aws_use_iam and database_passwordless_aws_region variables to runtime_container_engine_config module
- Configure DATABASE_AUTH_USE_AWS_IAM and DATABASE_AUTH_AWS_DB_REGION environment variables in database config
- Enable AWS IAM database authentication when enabled

raviharshicorp committed:
Add AWS IAM database authentication support

- Add database_passwordless_aws_use_iam and database_passwordless_aws_region variables
- Configure DATABASE_AUTH_USE_AWS_IAM and DATABASE_AUTH_AWS_DB_REGION environment variables
- Enable AWS IAM passwordless authentication for TFE runtime containers

raviharshicorp committed:
Fix duplicate variable declarations for AWS IAM database authentication

- Remove duplicate database_passwordless_aws_use_iam variable
- Remove duplicate database_passwordless_aws_region variable
- Variables are now declared only once as intended

tauhid621 commented on modules/runtime_container_engine_config/database_config.tf:
```diff
@@ -14,6 +14,8 @@ locals {
     TFE_DATABASE_CLIENT_KEY_FILE              = var.database_client_key_file
     TFE_DATABASE_PASSWORDLESS_AZURE_USE_MSI   = var.database_passwordless_azure_use_msi
     TFE_DATABASE_PASSWORDLESS_AZURE_CLIENT_ID = var.database_passwordless_azure_client_id
+    DATABASE_AUTH_USE_AWS_IAM                 = var.database_passwordless_aws_use_iam
```
Why are the variables not starting with `TFE` here?.
We should have consistency with other variables

raviharshicorp committed:
Fix: Add missing TFE_DATABASE_PASSWORDLESS_AWS_USE_INSTANCE_PROFILE for postgres passwordless

This critical fix adds the missing TFE_ prefixed environment variables that
the Go config system requires to properly configure AWS RDS IAM authentication.

Without these variables, the terraform-enterprise Go application cannot
read the passwordless configuration via envconfig, causing 502 errors.

Added:
- TFE_DATABASE_PASSWORDLESS_AWS_USE_INSTANCE_PROFILE (for Go config system)
- TFE_DATABASE_PASSWORDLESS_AWS_REGION (for Go config system)

Kept existing:
- DATABASE_AUTH_USE_AWS_IAM (for Atlas Ruby application)
- DATABASE_AUTH_AWS_DB_REGION (for Atlas Ruby application)

This matches the pattern used in redis_config.tf and ensures both
configuration systems receive the required environment variables.

raviharshicorp committed:
Remove Redis AWS IAM passwordless variables from PostgreSQL-only branch

raviharshicorp committed:
Remove all Redis AWS IAM passwordless configurations from PostgreSQL-only branch

- Remove Redis AWS IAM variables from runtime_container_engine_config
- Remove Redis AWS IAM variables from settings module
- Remove Redis AWS IAM configuration from tfe_redis_config.tf
- Keep only Azure MSI Redis variables that were in main branch
- This branch should only contain PostgreSQL passwordless authentication

raviharshicorp committed:
Complete cleanup of Redis AWS IAM references from redis_config.tf

raviharshicorp committed:
Add GCP IAM database authentication support

- Add database_passwordless_gcp_use_default_credentials variable
- Add DATABASE_AUTH_USE_GCP_IAM environment variable configuration
- Required for GCP postgres passwordless authentication in terraform-google-terraform-enterprise
